prow/plank/controller.go

```diff
@@ -524,41 +516,39 @@ func (c *Controller) syncTriggeredJob(pj prowapi.ProwJob, pm map[string]corev1.P
 	return c.prowJobClient.Patch(c.ctx, pj.DeepCopy(), ctrlruntimeclient.MergeFrom(&prevPJ))
 }
 
-// TODO: No need to return the pod name since we already have the
-// prowjob in the call site.
-func (c *Controller) startPod(pj prowapi.ProwJob) (string, string, error) {
+func (c *Controller) startPod(pj *prowapi.ProwJob) error {
 	buildID, err := c.getBuildID(pj.Spec.Job)
 	if err != nil {
-		return "", "", fmt.Errorf("error getting build ID: %v", err)
+		return fmt.Errorf("error getting build ID: %v", err)
 	}
 
-	pod, err := decorate.ProwJobToPod(pj, buildID)
+	pj.Status.BuildID = buildID
+	pod, err := decorate.ProwJobToPod(*pj)
 	if err != nil {
-		return "", "", err
+		return err
 	}
 	pod.Namespace = c.config().PodNamespace
 
 	client, ok := c.buildClients[pj.ClusterAlias()]
 	if !ok {
-		return "", "", fmt.Errorf("unknown cluster alias %q", pj.ClusterAlias())
+		return fmt.Errorf("unknown cluster alias %q", pj.ClusterAlias())
 	}
 	err = client.Create(c.ctx, pod)
-	c.log.WithFields(pjutil.ProwJobFields(&pj)).Debug("Create Pod.")
+	c.log.WithFields(pjutil.ProwJobFields(pj)).Debug("Create Pod.")
 	if err != nil {
-		return "", "", err
+		return err
 	}
-	return buildID, pod.ObjectMeta.Name, nil
+	pj.Status.PodName = pod.ObjectMeta.Name
+	return nil
 }
 
 func (c *Controller) getBuildID(name string) (string, error) {
 	return pjutil.GetBuildID(name, c.totURL)
 }
 
 func getPodBuildID(pod *corev1.Pod) string {
-	for _, env := range pod.Spec.Containers[0].Env {
-		if env.Name == "BUILD_ID" {
-			return env.Value
-		}
+	if buildID, ok := pod.ObjectMeta.Labels[kube.ProwBuildIDLabel]; ok {
```

Comment:
Since we've been failing to set this label, this won't work for existing jobs. We should ensure backwards compatibility.

Reply:
Thank you for the review. Fixed and added unit tests to cover.